.github/workflows/publish-to-PyPI.yml

```diff
@@ -1,18 +1,127 @@
+# GitHub Actions Workflow: Publish to PyPI
+# Description:
+# This workflow automates the process of testing and publishing the `py-marqo` package to PyPI.
+# The process includes:
+# 1. Starting a self-hosted EC2 runner.
+# 2. Running the `py-marqo` test suite on this runner.
+# 3. Stopping the EC2 runner.
+# 4. Publishing the package to PyPI if all previous steps were successful.
+
 name: Publish to PyPI
 
 on:
   push:
     branches:
       - mainline
+      - 'releases/*'
 
 permissions:
   contents: read
 
 jobs:
-  deploy:
+  Start-Runner:
+    name: Start self-hosted EC2 runner
+    runs-on: ubuntu-latest
+    outputs:
+      label: ${{ steps.start-ec2-runner.outputs.label }}
+      ec2-instance-id: ${{ steps.start-ec2-runner.outputs.ec2-instance-id }}
+    steps:
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_EC2_GH_RUNNER_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_EC2_GH_RUNNER_SECRET_ACCESS_KEY }}
+          aws-region: ${{ secrets.AWS_REGION }}
+      - name: Start EC2 runner
+        id: start-ec2-runner
+        uses: machulav/ec2-github-runner@v2
+        with:
+          mode: start
+          github-token: ${{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
+          ec2-image-id:  ${{ secrets.AMD_EC2_IMAGE_ID_200GB }}
+          ec2-instance-type: t3.xlarge
+          subnet-id: ${{ secrets.AMD_SUBNET_ID }}
+          security-group-id: ${{ secrets.AMD_SECURITY_GROUP_ID }}
+
+  Test-Py-Marqo:
```

Comment:
To lessen the redundant code, this would be a perfect place to use `called workflows`, and call the unit test workflow from here.